Eevee.Sleep.Bot/Modules/SlashCommands/ChangeRoleSlashModule.cs

```diff
@@ -0,0 +1,64 @@
+using Discord;
+using Discord.Interactions;
+using JetBrains.Annotations;
+using Eevee.Sleep.Bot.Controllers.Mongo;
+using Eevee.Sleep.Bot.Utils;
+
+namespace Eevee.Sleep.Bot.Modules.SlashCommands;
+
+[Group("role", "Commands for role things.")]
+public class ChangeRoleSlashModule : InteractionModuleBase<SocketInteractionContext>
+{
+    [SlashCommand("change", "Select a specific one from the role you have")]
+    [UsedImplicitly]
+    public async Task ChangeRoleAsync()
+    {
+        var roles = UserRoleContoller.FindRoleIdsByUserId(Context.User.Id.ToString());
+        var builder = new ComponentBuilder();
+
+        foreach(var role in roles)
+        {
+            builder.WithButton(
+                label: Context.Guild.Roles.Where(x => x.Id == role).Select(x => x.Name).First(), 
+                customId: $"role{role}"
+            );
+        }
+
+        await Context.Interaction.RespondAsync(
+            "Select your role.",
+            components: builder.Build(),
+            ephemeral: true
+        );
+    }
+
+    [SlashCommand("remove", "Remove role from DB")]
+    [RequireOwner]
+    [UsedImplicitly]
+    public async Task RemoveRoleAsync(IUser user, IRole role)
+    {
+        var roles = UserRoleContoller.RemoveRoles(user.Id.ToString(), [role.Id]);
+
+        await Context.Interaction.RespondAsync(
+            "Role Deleted.",
+            ephemeral: true
+        );
+    }
+
+    [SlashCommand("save", "Save all users roles")]
```

Comment:
Not sure what this does, is this to put all the roles a user have onto them?

Reply:
Then it sounds like it should be another command that allows adding/removing the role to track

Reply:
Already answered here
https://github.com/RaenonX-PokemonSleep/eevee-sleep-discord-bot/pull/25#discussion_r1609975652